### ISSUE TITLE
eth_getFilterChanges don't receive logs in pending state when toBlock: 'pending'

### ISSUE THREAD

**Issue by frozeman** [ethereum/go-ethereum#852]
Following scenario (use the browser, as the console doens't support filters yet):
- create contract
- create filter with toBlock: 'pending' to watch for the Deposit event
- send money to the contract
- Result: callback is only fired when block is mined, not when transaction is add to the pending state

``` js
walletABI = [   {      "anonymous" : false,      "inputs" : [],      "name" : "Created",      "type" : "event"   },   {      "anonymous" : false,      "inputs" : [         {            "indexed" : true,            "name" : "from",            "type" : "address"         },         {            "indexed" : false,            "name" : "value",            "type" : "uint256"         }      ],      "name" : "Deposit",      "type" : "event"   }];

walletABICompiled = '5b7f9adeddf84386b336eb7b3e18e7a6099be08fd81ea5d5142f4d2b630f8d20cf016040604090036040a15b60528060386000396000f300604c5b3373ffffffffffffffffffffffffffffffffffffffff167fe1fffcc4923d04b559f4d29a8bfc6cda04eb5b0d3c460751c2402c5c5cc9109c6040348152602001604090036040a25b565b60006000f3';
WalletContract = web3.eth.contract(walletABI);

// create contract
var contractInstance = new WalletContract({data: walletABICompiled, from: web3.eth.coinbase});

// watch pending tx (will fire, when the tx below are mined, not when add to the pending state!)
contractInstance.Deposit({}, {fromBlock: 0, toBlock: 'pending'}).watch(function(err,res){
       console.log('Deposit', res);
})

// send multiple tx to the address
web3.eth.sendTransaction({from: web3.eth.coinbase, to: contractInstance.address, value: web3.toWei(1, 'ether')});

web3.eth.sendTransaction({from: web3.eth.coinbase, to: contractInstance.address, value: web3.toWei(1, 'ether')});

web3.eth.sendTransaction({from: web3.eth.coinbase, to: contractInstance.address, value: web3.toWei(1, 'ether')});
```

**Comment by frozeman** [comment]
As discussed pending logs should return the the following fields as `null`:

- `blockNumber`
- `blockHash`

**Comment by tgerring** [comment]
Does this relate to #884 ?

**Comment by frozeman** [comment]
Nope. This one is about logs aren't being returned at all, #884 is about some properties should be `null`

**Comment by obscuren** [comment]
Is this still an issue @frozeman

**Comment by frozeman** [comment]
Last time it tested, yes.

**Comment by heikoheiko** [comment]
Is this still an issue? Would be a very helpful feature for more responsive DApps.

**Comment by frozeman** [comment]
@bas-vk

**Comment by berndbohmeier** [comment]
It is still an issue. 
If anyone facing the same problem, current workaround: this is working on parity;)

**Comment by frozeman** [comment]
@bas-vk is working on it.

**Comment by fjl** [comment]
See #3219 for WIP code.

**Comment by fjl** [comment]
Fixed in #3219
